src/main/java/com/autotune/common/datasource/DataSourceCollection.java

```diff
@@ -144,6 +144,10 @@ public void addDataSourcesFromConfigFile(String configFileName) {
                     DataSourceInfo dataSourceInfo = new ExperimentDBService().loadDataSourceFromDBByName(name);
                     if (null != dataSourceInfo) {
                         LOGGER.error("Datasource: {} already exists!", name);
+                        // add the auth details to local object
+                        AuthenticationConfig authConfig = getAuthenticationDetails(dataSourceObject, name);
```

Comment:
Do we need to check if getAuthenticationConfig() will retun null 
if so then do setAuthenticationConfig() ?

Reply:
We don't need that because in case of missing auth, `getAuthenticationDetails()` will return the default auth i.e. `noAuth`
Reference: https://github.com/kruize/autotune/blob/a2c4e341bc291696812589ed84b7709579cfd7b1/src/main/java/com/autotune/common/datasource/DataSourceCollection.java#L188